--- gensrc/parser.hpp	2015-05-09 11:48:36.219555956 +0200
+++ gensrc/parser.hpp~new	2015-05-09 12:11:00.406454334 +0200
@@ -200,7 +200,7 @@
     T&
     as ()
     {
-      YYASSERT (yytname_ == typeid (T).name ());
+      YYASSERT (!strcmp(yytname_, typeid (T).name ()));
       YYASSERT (sizeof (T) <= S);
       return *yyas_<T> ();
     }
@@ -210,7 +210,7 @@
     const T&
     as () const
     {
-      YYASSERT (yytname_ == typeid (T).name ());
+      YYASSERT (!strcmp(yytname_, typeid (T).name ()));
       YYASSERT (sizeof (T) <= S);
       return *yyas_<T> ();
     }
@@ -228,7 +228,7 @@
     swap (self_type& other)
     {
       YYASSERT (yytname_);
-      YYASSERT (yytname_ == other.yytname_);
+      YYASSERT (!strcmp(yytname_,other.yytname_));
       std::swap (as<T> (), other.as<T> ());
     }
 
